src/display/network.js

```diff
@@ -273,11 +273,20 @@ class PDFNetworkStreamFullRequestReader {
     const fullRequestXhrId = this._fullRequestId;
     const fullRequestXhr = this._manager.getRequestXhr(fullRequestXhrId);
 
-    const getResponseHeader = name => fullRequestXhr.getResponseHeader(name);
+    const responseHeaders = new Headers(
+      fullRequestXhr
+        .getAllResponseHeaders()
+        .trim()
+        .split("\r\n")
+        .map(x => {
+          const [key, ...val] = x.split(": ");
+          return [key, val.join(": ")];
+        })
```

Comment:
This is (effectively) just a slightly more compact version of https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest/getAllResponseHeaders#examples

Reply:
Is this fully equivalent? The MDN article uses `.split(/[\r\n]+/)` but here we use `.split("\r\n")` and AFAICT that has ever so slightly different behavior: the regex should match `\r` or `\n` as single characters, while the non-regex version should only match `\r\n` as combination. From the web console:

```
> "a\nb".split(/[\r\n]+/)
Array [ "a", "b" ]

> "a\nb".split("\r\n")
Array [ "a\nb" ]
```

If this is indeed not equivalent, we should probably also add a test case for this because the current ones then didn't cover it.

Reply:
At the top of https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest/getAllResponseHeaders it states the following, hence I figured that a *small* simplification might be OK?

> The [`XMLHttpRequest`](https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest) method `getAllResponseHeaders()` returns all the response headers, separated by [CRLF](https://developer.mozilla.org/en-US/docs/Glossary/CRLF), as a string, or returns `null` if no response has been received.

If you think that we should be more defensive here and thus use a regular expression that's obviously fine, but I cannot tell if that's actually necessary?

Reply:
Given that the text indeed explicitly says "separated by CRLF", which should indeed only be `\r\n`, I also think we don't have to change this then. I'm not really sure why the example there is more permissive then, but we can always change this later if need be.